### PR TITLE
Branch add remark feature

### LINKED ISSUE #46: Remark feature
Add a note to each patient profile so that a user is able to keep track of important information relating to patient care

### PR DESCRIPTION
**Pull request by volleyballkickedme**
Add remark feature so that a user is able to view additional information regarding a patient's care status

close #46

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_REMARK + "[REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_REMARK + "Likes to swim.";
+
+    public static final String MESSAGE_ARGUMENTS = "Index: %1$d, Remark: %2$s";
+
+    private final Index index;
+    private final Remark remark;
+    public static final String MESSAGE_ADD_REMARK_SUCCESS = "Added remark to Person: %1$s";
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:30]: maybe group similar variables together in the code for readability? eg. all "MESSAGE_" variables in one place to distinguish groups of variables appropriately

File: src/main/java/seedu/address/model/person/Remark.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package seedu.address.model.person;
+import static java.util.Objects.requireNonNull;
+/**
+ * Represents a Person's remark in the address book.
+ * Guarantees: immutable; is always valid
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/model/person/Remark.java:5]: maybe not an apt comment for the javadocs?

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json [hunk 3]
```diff
@@ -5,42 +5,49 @@
     "phone" : "94351253",
     "email" : "alice@example.com",
     "address" : "123, Jurong West Ave 6, #08-111",
+    "remark" : "",
```
Comment (itsnotsherm) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:9]: if remarks are usually empty initially I was thinking can explore having it as an optional parameter, can look into this further

File: src/test/java/seedu/address/logic/commands/RemarkCommandTest.java
Comment (itsnotsherm) [inline src/test/java/seedu/address/logic/commands/RemarkCommandTest.java:1]: thanks for adding test cases for the remark command

**Review by itsnotsherm — APPROVED** [review]
LGTM, remember to fix checkstyle before merging

**Review by reidenong — APPROVED** [review]
Looks good, tests are comprehensive as well.

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json (hunk 3 above)
Comment (reidenong) [reply src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:9]: I think for something like remarks a safe implementation is to always have the field there, maybe we can choose to fix this on display side (Don't display empty remarks on the GUI). Reason being its probably not worth the hassle to deal with null values or Optionals.

**Pull request merged**